internal/model/rule.go

```diff
@@ -58,6 +67,14 @@ type RuleTF struct {
 	LastUpdated types.String `tfsdk:"-"`
 }
 
+func (r RuleTF) TFIdentifier() string {
+	if r.Segment.IsNull() {
+		return r.Metric.ValueString()
+	}
+
+	return fmt.Sprintf("%s/%s", r.Segment.ValueString(), r.Metric.ValueString())
```

Comment:
could a segment string contain a `/`?

Reply:
Segment ids are ULIDs, so no. I found a better way to do this though, so I'll send another PR once all these stacked ones are merged which cleans this up.